AuthSAML.php

```diff
@@ -118,9 +125,23 @@ public function getGlobalBasePermissions()
         ));
     }
 
-    public function beforeLogin() {
+    /**
+     * @see event
+     * Set the auth if needed, connect if authenticated
+     **/
+    public function beforeLogin()
+    {
+        if (App()->getSession()->get('auth_saml_invalid')) {
```

Comment:
We noticed that: when a user tries to login without SAML (relevant "Force SAML login" setting is OFF), he fails. He has to try the SAML login and after this one fails, then he can login normally as it could before. I think this IF block likely causes that behavior.

Reply:
No real reason here.
Since App()->getSession()->get('auth_saml_invalid') retuen null if not set.

Reply:
https://github.com/Shnoulle/Limesurvey-SAML-Authentication/issues/1

I try some debug, i don't understand : semms AuthDB didn't do the trick